### PR TITLE
Player.java Move listeners to internal classes

### PR DESCRIPTION
**Pull request by evermind-zz**
For me it was at first not so easy  to understand which method belongs to which Interface in Player.java.

So I decided to move methods implementing interfaces into their own innner classes. In my opinion this will increase the readability for new developers. 

The patches are from May but I rebased them today!

**

#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- create innner classes for each Listener
- increase readabilty

#### APK testing 

On the website the APK can be found by going to the "Checks" tab below the title and then on "artifacts" on the right.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/player/Player.java [hunk 1]
```diff
@@ -401,6 +401,17 @@ public Player(@NonNull final MainPlayer service) {
...
+        playbackStateListener = new PlaybackStateListener();
+        playerPlaybackListener = new PlayerPlaybackListener();
+        playerImageLoaderListener = new PlayerImageLoadingListener();
+        playerVideoListener = new PlayerVideoListener();
+
+        playerOnSeekBarChangeListener = new PlayerOnSeekBarChangeListener();
+        playerViewOnClickListener = new PlayerViewOnClickListener();
+        playerViewOnLongClickListener = new PlayerViewOnLongClickListener();
+        playerPopupMenuOnMenuItemClickListener = new PlayerPopupMenuOnMenuItemClickListener();
+        playerPopupMenuOnDismissListener = new PlayerPopupMenuOnDismissListener();
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/Player.java:414]: Just initialize them directly in the `class` body.

**Review by Stypox — COMMENTED** [review]
I don't know if I want this merged. When I ordered the functions I used some semantically chosen categories. What you are proposing here is, instead, choosing categories just based on what listeners the functions belong to, not what they do and what they mean. So here the click listeners are all over the place, and they are more difficult to navigate in my opinion.

**Comment by evermind-zz** [comment]
> I don't know if I want this merged. When I ordered the functions I used some semantically chosen categories. What you are proposing here is, instead, choosing categories just based on what listeners the functions belong to, not what they do and what they mean. So here the click listeners are all over the place, and they are more difficult to navigate in my opinion.

Yes, I saw the categories and I understand the idea, But I still felt lost inside this big class. Yes all related methods should also be moved inside the corresponding inner listener class. I'll update the pull request.

File: app/src/main/java/org/schabi/newpipe/player/Player.java (hunk 1 above)
Comment (evermind-zz) [reply app/src/main/java/org/schabi/newpipe/player/Player.java:414]: will change that

**Comment by evermind-zz** [comment]
Maybe its not worth at all. Maybe we just could fix the region displaying correctly in android studio. For example changing

`--- a/app/src/main/java/org/schabi/newpipe/player/Player.java`
`+++ b/app/src/main/java/org/schabi/newpipe/player/Player.java`
`@@ -4038,10 +4038,9 @@ public final class Player implements`
`     //endregion`
` `
` `
`-    /*//////////////////////////////////////////////////////////////////////////`
`-    // Getters`
`-    //////////////////////////////////////////////////////////////////////////*/`
`-    //region`
`+    //////////////////////////////////////////////////////////////////////////`
`+    //region Getters`
`+    //////////////////////////////////////////////////////////////////////////`

Nevertheless I've updated the pull request with moving the initialization  to the body

**Comment by Stypox** [comment]
@evermind-zz Oh, I didn't know you could use labels with `//region`. I like your solution, that looks readable both in IDEs with region code folding (Android Studio) and others (e.g. GitHub).

**Comment by Stypox** [comment]
Any update?

**Comment by evermind-zz** [comment]
> Any update?

I'll did not yet had the time to do it! I will try to find some time

**Pull request closed without merging**